### ISSUE TITLE
Enhancement for clang-tidy readability-redundant-string-cstr check 

### ISSUE THREAD

**Issue by nunema** [llvm/llvm-project#59255]
The readability-redundant-string-cstr warning should also fire when constructing `string_view` and `filesystem::path`.

```cpp
#include <string>
#include <filesystem>

// readability-redundant-string-cstr

int
main()
{
  std::string s ("foo");

  // This does warn
  std::string t (s.c_str());

  // These should warn, but do not.
  std::string_view u (s.c_str());
  std::filesystem::path v (s.c_str());

  (void)t;
  (void)u;
  (void)v;

  return 0;
}
```

**Comment by deeptirajora** [comment]
i think it should be like this :

#include <string>
#include <filesystem>
int main()
{
    std::string s("foo");
    std::string t(s); 
    std::string_view u(s);
    std::filesystem::path v(s);
    (void)t;
    (void)u;
    (void)v;
    return 0;
}

**Comment by EugeneZelenko** [comment]
Duplicate of #25244.
